src/Microsoft.NetCore.Analyzers/Core/Security/DoNotUseInsecureCryptographicAlgorithms.cs

```diff
@@ -78,13 +78,105 @@ public override void Initialize(AnalysisContext analysisContext)
+                                messageArgs[1] = cryptTypes.HMACSHA1.Name;
+                            }
+                            else if (type.DerivesFrom(cryptTypes.DES))
+                            {
+                                rule = DoNotUseBrokenCryptographyRule;
+                                messageArgs[1] = cryptTypes.DES.Name;
+                            }
+                            else if ((method.ContainingType.DerivesFrom(cryptTypes.DSA)
+                                      && method.MetadataName == SecurityMemberNames.CreateSignature)
+                                || (type == cryptTypes.DSASignatureFormatter
+                                    && method.ContainingType.DerivesFrom(cryptTypes.DSASignatureFormatter)
+                                    && method.MetadataName == WellKnownMemberNames.InstanceConstructorName))
+                            {
+                                rule = DoNotUseBrokenCryptographyRule;
+                                messageArgs[1] = cryptTypes.DSA.Name;
+                            }
+                            else if (type.DerivesFrom(cryptTypes.HMACMD5))
+                            {
+                                rule = DoNotUseBrokenCryptographyRule;
+                                messageArgs[1] = cryptTypes.HMACMD5.Name;
+                            }
+                            else if (type.DerivesFrom(cryptTypes.RC2))
+                            {
+                                rule = DoNotUseBrokenCryptographyRule;
+                                messageArgs[1] = cryptTypes.RC2.Name;
+                            }
+                            else if (type.DerivesFrom(cryptTypes.TripleDES))
+                            {
+                                rule = DoNotUseWeakCryptographyRule;
+                                messageArgs[1] = cryptTypes.TripleDES.Name;
+                            }
+                            else if (type.DerivesFrom(cryptTypes.RIPEMD160))
+                            {
+                                rule = DoNotUseWeakCryptographyRule;
+                                messageArgs[1] = cryptTypes.RIPEMD160.Name;
+                            }
+                            else if (type.DerivesFrom(cryptTypes.HMACRIPEMD160))
+                            {
+                                rule = DoNotUseWeakCryptographyRule;
+                                messageArgs[1] = cryptTypes.HMACRIPEMD160.Name;
+                            }
+
+                            if (rule != null)
+                            {
+                                operationAnalysisContext.ReportDiagnostic(
+                                    Diagnostic.Create(
+                                        rule,
+                                        operationAnalysisContext.Operation.Syntax.GetLocation(),
+                                        messageArgs));
+                            }
+
```

Comment:
Nit: remove additional blank line. @sharwell I'd thought we would get a formatting violation (IDE0055) for this one?

Reply:
No, the formatter doesn't care about blank lines.